src/trans.rs

```diff
@@ -719,6 +727,48 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
                 }
             }
 
+            Rvalue::CheckedBinaryOp(ref op, ref left, ref right) => {
+                // TODO: This shouldn't just be a copy BinaryOp above!
+                // We should do some actual _checking_!
```

Comment:
I was actually not thinking of writing it in rust, though that's also a good idea.  The problem I'm trying to avoid is that checked operations tend to do a number on the local control flow - and I haven't fully wrapped my head around the binaryen APIs needed to implement that.  Just emitting a `call` makes this really easy.  But I could easily be dramatically overestimating the complexity of just doing it the obvious (inline) way.